Sources/SourceKitLSP/Swift/DocumentFormatting.swift

```diff
@@ -170,9 +202,10 @@ extension SwiftLanguageService {
       swiftFormatConfiguration(for: textDocument.uri, options: options),
     ]
     if let range {
+      let utf8Range = snapshot.utf8OffsetRange(of: range)
       args += [
         "--offsets",
-        "\(snapshot.utf8Offset(of: range.lowerBound)):\(snapshot.utf8Offset(of: range.upperBound))",
+        "\(utf8Range.lowerBound):\(utf8Range.upperBound - 1)",
```

Comment:
swift-format takes an inclusive range, but Swift's `Range.upperBound` is exclusive.

Reply:
I reckon this will go wrong if `lowerBound == upperBound`? do i just manually handle that with some kind of if statement, or is there a better way?

Reply:
Nice catch. Can you add the explanation that swift-format takes an inclusive range to the code?

I would add a special case for `lowerBound == upperBound`.